portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,16 +44,43 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String name = getName(request);
     String comment = getComment(request);
+
     // Prevents users from submitting comments with no content."
     if (comment.equals("")) {
         response.setContentType("text/html");
         response.getWriter().println("Please enter a comment");
       return;
     }
-    comments.add(new Comment(comment, name));
+    // Store new comment into datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
I notice that you connect here and in the get - a common pattern is to initialize your database connection in a constructor or a factory so that you don't need to re-connect on every call. This can make error handling and parallelization easier down the line. For Java specifically I don't know if a constructor or factory is recommended but either works for me.

Reply:
That makes sense--it's actually one of the questions I noted down in my daily reflection doc. I'll put it in the constructor.